lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -447,6 +455,9 @@ public void OnNext(ICompletedTask completedTask)
                         _taskManager.RecordCompletedTask(completedTask);
                         TryRecovery();
                         break;
+                    case SystemState.Fail:
+                        FailAction();
```

Comment:
Right after we change the system state to Fail, we have called FailAction() in the same lock. So if the system state is already Fail, we don't need to call FailAction() any more.